jetstream/statistics.py

```diff
@@ -116,23 +116,25 @@ def apply(self, df: DataFrame, metric: str) -> "StatisticResultCollection":
 
         if metric in df:
             branch_list = df.branch.to_numpy()
-            if self.ref_branch_label not in branch_list:
-                logging.warn(
-                    f"Branch {self.ref_branch_label} not in {branch_list} for {self.name()}."
-                )
+            if reference_branch not in branch_list or reference_branch is None:
+                logging.warn(f"Branch {reference_branch} not in {branch_list} for {self.name()}.")
             else:
-                statistic_result_collection.data += self.transform(df, metric).data
+                statistic_result_collection.data += self.transform(
+                    df, metric, reference_branch
+                ).data
 
         return statistic_result_collection
 
     @abstractmethod
-    def transform(self, df: DataFrame, metric: str) -> "StatisticResultCollection":
+    def transform(
+        self, df: DataFrame, metric: str, reference_branch: str
+    ) -> "StatisticResultCollection":
         return NotImplemented
 
     @classmethod
     def from_dict(cls, config_dict: Dict[str, Any]):
         """Create a class instance with the specified config parameters."""
-        return cls(**config_dict)
+        return cls(**config_dict)  # type: ignore
```

Comment:
mypy is complaining about this one...

Reply:
I added and then removed a patch that called `cattr.structure(config_dict, cls)` instead. I think silencing mypy is probably better here.

This is perfectly legal:
```
class A:
    pass

A(**{})
```

It's always true that instantiating a class from kwargs may fail if the class doesn't understand the keys of the dict. If the constructor doesn't accept any non-self arguments at all, the same principle applies. I could be wrong but I think we just disagree with mypy.

The alternative (`cattr.structure`) just ignores keys that the class doesn't understand. That makes the operation type-safe but I think we probably shouldn't let that pass silently; throwing an exception seems better.

So this seems right to me.